src/utils/CiceroMarkToOOXML/index.js

```diff
@@ -0,0 +1,155 @@
+const insertHeading = async (context, value, level) => {
+  const definedLevels = {
+    1: 28,
+    2: 26,
+    3: 24,
+    4: 22,
+    5: 20,
+    6: 18,
+  };
+  context.document.body.insertParagraph(value, Word.InsertLocation.end).font.set({
+    color: 'black',
+    bold: false,
+    size: definedLevels[level],
+  });
+  insertLineBreak(context);
+  await context.sync();
+};
+
+const insertLineBreak = async context => {
+  context.document.body.insertBreak(Word.BreakType.line, Word.InsertLocation.end);
+  context.sync();
+};
+
+const insertSoftBreak = async context => {
+  context.document.body.insertText(' ', Word.InsertLocation.end);
+  await context.sync();
+};
+
+const insertText = async (context, value) => {
+  context.document.body.insertText(value, Word.InsertLocation.end).font.set({
+    bold: false,
+    color: 'black',
+    size: 14,
+  });
+  await context.sync();
+};
+
+const insertVariable = async (context, title, tag, value) => {
+  let variableText = context.document.body.insertText(value, Word.InsertLocation.end);
+  let contentControl = variableText.insertContentControl();
+  contentControl.title = title;
+  contentControl.tag = tag;
+  contentControl.font.set({
+    color: 'black',
+    bold: true,
+    size: 14,
+  });
+  await context.sync();
+
+  setTimeout(() => {
+    Office.context.document.bindings.addFromNamedItemAsync(title, 'text', { id: title }, res => {
+      if (res.status === Office.AsyncResultStatus.Succeeded) {
+        res.value.addHandlerAsync(Office.EventType.BindingDataChanged, handler);
+      }
+      else {
+        // ToDo: show the error to user in Production environment
+        console.error(title, res);
+      }
+    });
+  }, 100);
```

Comment:
1. Adding the variable text (eg. Party A).
2. Wrapping it around content control.

These two processes takes time to sync with the Word document.
Therefore, I have added a `setTimeout` to ensure the listener is added when the text has been synced for sure.

Since this is a workaround, I plan to remove this soon. I do have a better idea to do so.

Reply:
Can you `await` `Office.context.document.bindings.addFromNamedItemAsync` and `res.value.addHandlerAsync` instead of doing the setTimeout? That way it will finish before moving on to the sync.